src/celeritas/optical/ScintillationData.hh

```diff
@@ -29,67 +35,151 @@ struct ScintillationComponent
  * number of photons generated.
+ * \c components stores the fast/slow/etc scintillation components for this
+ * material.
  */
-struct ScintillationSpectrum
+struct MaterialScintillationSpectrum
 {
     real_type yield{};
-    real_type resolution_scale{};
     ItemRange<ScintillationComponent> components;
 
     //! Whether all data are assigned and valid
     explicit CELER_FUNCTION operator bool() const
     {
-        return yield > 0 && resolution_scale >= 0 && !components.empty();
+        return yield > 0 && !components.empty();
+    }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Data characterizing the scintillation spectrum for a given particle in a
+ * given material.
+ *
+ * \c yield_vector is the characteristic light yield for different energies.
+ * \c components stores the fast/slow/etc scintillation components for this
+ * particle type.
+ */
+struct ParticleScintillationSpectrum
+{
+    GenericGridData yield_vector;
+    ItemRange<ScintillationComponent> components;
+
+    //! Whether all data are assigned and valid
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return static_cast<bool>(yield_vector);
     }
 };
 
 //---------------------------------------------------------------------------//
 /*!
- * Scintillation data tabulated with the optical material id.
+ * Data characterizing the scintillation spectrum for all particles and
+ * materials.
+ *
+ * Sampling using material-only data or particle- and material-dependent data
+ * are mutually exclusive. Therefore, either \c materials or \c particles are
+ * loaded at the beginning of the simulation, but *never* both at the same
+ * time. The \c scintillation_by_particle() function can be used to check that.
+ *
+ * - \c matid_to_optmatid returns an \c OpticalMaterialId given a
+ *   \c MaterialId
+ * - \c pid_to_scintpid returns a \c ScintillationParticleId given a
+ *   \c ParticleId .
+ * - \c resolution_scale is indexed by \c OpticalMaterialId .
+ * - \c materials stores material-only scintillation data. Indexed by
+ *   \c OpticalMaterialId
+ * - \c particles stores scintillation data for each particle type for each
+ *   material, being a grid of size `num_particles * num_materials`. Therefore
+ *   it is indexed by \c ParticleScintSpectrumId , which combines
+ *   \c ScintillationParticleId and \c OpticalMaterialId . Use the
+ *   \c spectrum_index() function to retrieve the correct index.
  */
 template<Ownership W, MemSpace M>
 struct ScintillationData
 {
     template<class T>
     using Items = Collection<T, W, M>;
-    template<class T>
-    using OpticalMaterialItems = Collection<T, W, M, OpticalMaterialId>;
+    using MaterialItems
+        = Collection<MaterialScintillationSpectrum, W, M, OpticalMaterialId>;
+    using ParticleItems
+        = Collection<ParticleScintillationSpectrum, W, M, ParticleScintSpectrumId>;
```

Comment:
```suggestion
    template<class T>
        using OpticalMaterialItems = Collection<T, W, M, OpticalMaterialId>;
    template<class T>
    using ParticleSpectrumItems
        = Collection<T, W, M, ParticleScintSpectrumId>;
```
and then use `OpticalMaterialItems<MaterialScintillationSpectrum>` and `OpticalMaterialItems< real_type >`  and `ParticleSpectrumItems < ParticleScintillationSpectrum>` (or even omit the typedef no the last one since it's unique to that indexing.